library/core/src/ptr/non_null.rs

```diff
@@ -308,6 +317,8 @@ impl<T: ?Sized> NonNull<T> {
     #[must_use]
     #[inline]
     #[unstable(feature = "strict_provenance", issue = "95228")]
+    //guarentees that function always returns a valid (non-null) pointer
+    #[ensures(|result: &Self| !result.as_ptr().is_null())]
     pub fn map_addr(self, f: impl FnOnce(NonZero<usize>) -> NonZero<usize>) -> Self {
```

Comment:
Do we need to add more pre-condition for `map_addr', didn't see in both function contracts and proof harness